cmd/replay/main.go

```diff
@@ -247,7 +247,9 @@ func main() {
 		// Espresso-specific validation
 		// TODO test: https://github.com/EspressoSystems/espresso-sequencer/issues/772
 		isL2Message := message.Message.Header.Kind == arbostypes.L1MessageType_L2Message
-		if isL2Message && chainConfig.Espresso {
+		commitment := espressoTypes.Commitment(wavmio.ReadHotShotCommitment(inboxPos, posInInbox))
+		// When the commitment is not empty, the currently validating message comes from Espresso
+		if isL2Message && commitment != [32]byte{} {
```

Comment:
Previously an empty hotshot commitment in espresso mode would fail validation, now it would pass. Can we be sure that the commitment will never be empty when it shouldn't be empty?

Reply:
Here I use the convention that `empty hotshot commitment` means we are not running in espresso mode. This code is in the `replay wasm` and every flag is passed into it by `wavmio`, so we need to make sure the `validator` or `arbitrator` is passing empty hotshot commitment into it when they are not in the espresso mode. [Here](https://github.com/EspressoSystems/nitro-espresso-integration/blob/7f7aeca70a1b24d3886152eec47dfe9ec7b6dd9e/staker/block_validator.go#L503C2-L540C49) is what we do in validator.
Previously this didn't pass because some tests (related to fraud proof I guess) couldn't read hotshot commitment using the `arbitrator machine`(not sure) and now I set a default empty hotshot commitment so it works now.